pkg/gitconfig/configs.go

```diff
@@ -0,0 +1,358 @@
+package gitconfig
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gopasspw/gopass/internal/set"
+	"github.com/gopasspw/gopass/pkg/appdir"
+	"github.com/gopasspw/gopass/pkg/debug"
+)
+
+// Configs is a container for a config "view" that is composed of several different
+// config objects. The intention is for the ones with a wider scope to provide defaults
+// so those with a more narrow scope then only have to override what they are interested in.
+type Configs struct {
+	Preset   *Config
+	system   *Config
+	global   *Config
+	local    *Config
+	worktree *Config
+	env      *Config
+	workdir  string
+
+	SystemConfig   string
+	GlobalConfig   string
+	LocalConfig    string
+	WorktreeConfig string
+	EnvPrefix      string
+	NoWrites       bool
+}
+
+func New() *Configs {
+	return &Configs{
+		system: &Config{
+			readonly: true,
+		},
+		global: &Config{
+			path: globalConfigFile(),
+		},
+		local:    &Config{},
+		worktree: &Config{},
+		env: &Config{
+			noWrites: true,
+		},
+
+		SystemConfig:   systemConfig,
+		GlobalConfig:   globalConfig,
+		LocalConfig:    localConfig,
+		WorktreeConfig: worktreeConfig,
+		EnvPrefix:      envPrefix,
+	}
+}
+
+// Reload will reload the config(s) from disk.
+func (cs *Configs) Reload() {
+	cs.LoadAll(cs.workdir)
+}
+
+// LoadAll tries to load all known config files. Missing or invalid files are
+// silently ignored. It never fails. The workdir is optional. If non-empty
+// this method will try to load a local config from this location.
+func (cs *Configs) LoadAll(workdir string) *Configs {
+	cs.workdir = workdir
+
+	debug.Log("Loading gitconfigs for %+v ...", cs)
+
+	// load the system config, if any
+	if os.Getenv(cs.EnvPrefix+"_NOSYSTEM") == "" {
+		c, err := LoadConfig(cs.SystemConfig)
+		if err != nil {
+			debug.Log("failed to load system config: %s", err)
+		} else {
+			debug.Log("loaded system config from %s", cs.SystemConfig)
+			cs.system = c
+			cs.system.readonly = true
```

Comment:
Not sure why the system config is `readonly`?

Reply:
Add some comment.